src/reanimated2/animation/styleAnimation.ts

```diff
@@ -33,43 +87,40 @@ export function withStyleAnimation(
       now: Timestamp
     ): boolean => {
       let stillGoing = false;
-      Object.keys(styleAnimations).forEach((key) => {
-        const currentAnimation = animation.styleAnimations[key];
-        if (key === 'transform') {
-          const transform = animation.styleAnimations.transform;
-          for (let i = 0; i < transform.length; i++) {
-            const type = Object.keys(transform[i])[0];
-            const currentAnimation: AnimationObject = transform[i][type];
-            if (currentAnimation.finished) {
-              continue;
-            }
-            const finished = currentAnimation.onFrame(currentAnimation, now);
-            if (finished) {
-              currentAnimation.finished = true;
-              if (currentAnimation.callback) {
-                currentAnimation.callback(true);
-              }
-            } else {
-              stillGoing = true;
-            }
-            // @ts-ignore: eslint-disable-line
-            animation.current.transform[i][type] = currentAnimation.current;
-          }
+      const onFrameRecursive = (
+        currentStyleAnimation: NestedObjectValues<AnimationObject>,
+        currentPath: PrimitiveValue[]
+      ): void => {
+        if (Array.isArray(currentStyleAnimation)) {
+          currentStyleAnimation.forEach((element, index) =>
+            onFrameRecursive(element, [...currentPath, index])
```

Comment:
I guess in order to make sure it works perfectly for all the envs and just as fast as possible `Array.concat` can be used here
```suggestion
            onFrameRecursive(element, currentPath.concat(index))
```